worker/uniter/op_callbacks.go

```diff
@@ -67,20 +65,8 @@ func (opc *operationCallbacks) PrepareHook(hi hook.Info) (string, error) {
 		name = fmt.Sprintf("%s-%s", storageName, hi.Kind)
 		// TODO(axw) if the agent is not installed yet,
 		// set the status to "preparing storage".
-	case hi.Kind == hooks.Stop:
-		// TODO(perrito666) I dont know if this should be Lost.
-		status = params.StatusExecuting
 	case hi.Kind == hooks.ConfigChanged:
 		opc.u.f.DiscardConfigEvent()
-		fallthrough
-	default:
-		if !opc.u.operationState().Started {
-			status = params.StatusExecuting
-		}
-	}
-	err := opc.u.unit.SetAgentStatus(status, "", nil)
-	if err != nil {
-		return "", err
 	}
```

Comment:
Yes, I've reworked it so that the executing status is set in the relevant operations implementation (run hook, run action, run commands) rather than in modes.